common/src/main/java/org/apache/cassandra/sidecar/common/data/Digest.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.common.data;
+
+import java.util.Map;
+import java.util.function.Consumer;
+
+/**
+ * Interface that represents a checksum digest
+ */
+public interface Digest extends Consumer<Map<String, String>>
```

Comment:
- The `Digest` interface is only used from the client side to put headers. Can you relocate the interface and its implementations to client package? 
- Despite `Consumer` does the job seemingly, can we have a more dedicated interface for digest?  

```java
Digest 
{
   // supply the header to be used in http request
   Map<String, String> headers();
}

// usage, e.g. in UploadSSTableRequest#headers() method, merge the headers with others
Map<String, String> headers = new HashMap<>(super.headers());
headers.putAll(digest.headers()); 
```

Reply:
we are now using it in both the client / server code. so leaving it in common for now